packages/menu/src/test/mwc-menu.test.ts

```diff
@@ -0,0 +1,307 @@
+  });
+
+  test('`fixed` is passed to surface', async () => {
+    const surface =
+        element.shadowRoot!.querySelector<MenuSurface>('.mdc-menu')!;
+    assert.equal(element.fixed, false);
+    assert.equal(surface.fixed, false);
+    element.fixed = true;
+    await element.updateComplete;
+    assert.equal(element.fixed, true);
+    assert.equal(surface.fixed, true);
+  });
+
+  test('`fullwidth` is passed to surface', async () => {
+    const surface =
+        element.shadowRoot!.querySelector<MenuSurface>('.mdc-menu')!;
+    assert.equal(element.fullwidth, false);
+    assert.equal(surface.fullwidth, false);
+    element.fullwidth = true;
+    await element.updateComplete;
+    assert.equal(element.fullwidth, true);
+    assert.equal(surface.fullwidth, true);
+  });
+
+  test('correct roles are passed to inner list', async () => {
+    const list = element.shadowRoot!.querySelector<List>('.mdc-list')!;
+    assert.equal(element.innerRole, 'menu');
+    assert.equal(list.innerRole, 'menu');
+    assert.equal(list.itemRoles, 'menuitem');
+    element.innerRole = 'listbox';
+    await element.updateComplete;
+    assert.equal(list.innerRole, 'listbox');
+    assert.equal(list.itemRoles, 'option');
+  });
+
+  test('`multi` is set on inner list', async () => {
+    const list = element.shadowRoot!.querySelector<List>('.mdc-list')!;
+    assert.equal(element.multi, false);
+    assert.equal(list.multi, false);
+    element.multi = true;
+    await element.updateComplete;
+    assert.equal(element.multi, true);
+    assert.equal(list.multi, true);
+  });
+
+  test('`activatable` is set on inner list', async () => {
+    const list = element.shadowRoot!.querySelector<List>('.mdc-list')!;
+    assert.equal(element.activatable, false);
+    assert.equal(list.activatable, false);
+    element.activatable = true;
+    await element.updateComplete;
+    assert.equal(element.activatable, true);
+    assert.equal(list.activatable, true);
+  });
+
+  suite('selection', () => {
+    setup(async () => {
+      element.innerHTML = `
+        <mwc-list-item>1</mwc-list-item>
+        <mwc-list-item>2</mwc-list-item>`;
+      element.layout(true);
+      await element.updateComplete;
+    });
+
+    test('`index` returns selected index', async () => {
+      const list = element.shadowRoot!.querySelector('mwc-list')!;
+      assert.equal(list.index, -1);
+      assert.equal(element.index, -1);
+      element.select(1);
+      await element.updateComplete;
+      assert.equal(list.index, 1);
+      assert.equal(element.index, 1);
+    });
+
+    test('`selected` returns selected item', async () => {
+      const list = element.shadowRoot!.querySelector('mwc-list')!;
+      const item = element.children[1] as ListItem;
+      element.select(1);
+      await element.updateComplete;
+      assert.equal(element.selected, item);
+      assert.equal(list.selected, item);
+    });
+  });
+});
+
+suite('mwc-menu-surface', () => {
```

Comment:
Missing cases:

* calling show method
* restore focus: focusing something, opening the surface, and focusing something inside the surface, then closing it and seeing if focus was restored

File a bug if you don't want to implement